src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs

```diff
@@ -839,48 +841,95 @@ private async Task<double> GetNetworkBlockAverageTime(PoolConfig poolConfig)
-                    if(transaction.HasErrors().GetValueOrDefault())
+                    BigInteger? nonce = null;
+                    // Use existing nonce to avoid duplicate transaction
+                    Nethereum.RPC.Eth.DTOs.TransactionReceipt txReceipt;
+                    if(transactionHashes.TryGetValue(balance.Address, out var prevTxHash))
                     {
-                        logger.Error($"[{LogCategory}] Web3Tx failed. status={transaction.Status}, addr={balance.Address}, amt={balance.Amount}");
+                        // Check if existing transaction was succeeded
+                        txReceipt = await TelemetryUtil.TrackDependency(() => txService.Transactions.GetTransactionReceipt.SendRequestAsync(prevTxHash),
+                            DependencyType.Web3, "GetTransactionReceipt", $"addr={balance.Address},amt={balance.Amount.ToStr()},txhash={prevTxHash}");
+                        
+                        if(txReceipt != null)
+                        {
+                            if(txReceipt.HasErrors().GetValueOrDefault())
+                            {
+                                logger.Error($"[{LogCategory}] Web3Tx failed without a receipt. addr={balance.Address},amt={balance.Amount.ToStr()},status={txReceipt.Status}");
+                                return null;
+                            }
+                            logger.Info($"[{LogCategory}] Web3Tx receipt found for existing tx. addr={balance.Address},amt={balance.Amount.ToStr()},txhash={prevTxHash},gasfee={txReceipt.EffectiveGasPrice}");
+
+                            transactionHashes.TryRemove(balance.Address, out _);
+                            return new TransactionReceipt
+                            {
+                                Id = txReceipt.TransactionHash,
+                                Fees = Web3.Convert.FromWei(txReceipt.EffectiveGasPrice),
+                                Fees2 = Web3.Convert.FromWei(txReceipt.GasUsed)
+                            };
+                        }
+                        // Get nonce for existing transaction
+                        var prevTx = await TelemetryUtil.TrackDependency(() => txService.Transactions.GetTransactionByHash.SendRequestAsync(prevTxHash),
+                            DependencyType.Web3, "GetTransactionByHash", $"addr={balance.Address},amt={balance.Amount.ToStr()},txhash={prevTxHash}");
+                        nonce = prevTx?.Nonce;
+                        logger.Info($"[{LogCategory}] Web3Tx receipt not found for existing tx. addr={balance.Address},amt={balance.Amount},txhash={prevTxHash},nonce={nonce}");
+                    }
+
+                    // Queue transaction
+                    txHash = await TelemetryUtil.TrackDependency(() => txService.GetEtherTransferService().TransferEtherAsync(balance.Address, balance.Amount, nonce: nonce),
```

Comment:
we're getting the nonce from the previous transaction and then sending the next transaction with the same nonce?

don't we need to increment it?

Reply:
We essentially resubmitting the transaction for the same address.